File: web/src/types/strategies.ts
export interface StrategyCoinSourceConfigPayload {
  source_type: string
  static_coins: string[]
  excluded_coins: string[]
  use_ai500: boolean
  ai500_limit: number
  use_oi_top: boolean
  oi_top_limit: number
  use_oi_low: boolean
  oi_low_limit: number
}

export interface StrategyKlinesConfigPayload {
  primary_timeframe: string
  primary_count: number
  longer_timeframe: string
  longer_count: number
  enable_multi_timeframe: boolean
  selected_timeframes: string[]
}

export interface StrategyIndicatorsConfigPayload {
  klines: StrategyKlinesConfigPayload
  enable_raw_klines: boolean
  enable_ema: boolean
  enable_macd: boolean
  enable_rsi: boolean
  enable_atr: boolean
  enable_boll: boolean
  enable_volume: boolean
  enable_oi: boolean
  enable_funding_rate: boolean
  quantauraos_api_key: string
  enable_quant_data: boolean
  enable_oi_ranking: boolean
  enable_netflow_ranking: boolean
  enable_price_ranking: boolean
}

export interface StrategyRiskControlConfigPayload {
  max_positions: number
  btc_eth_max_leverage: number
  altcoin_max_leverage: number
  btc_eth_max_position_value_ratio: number
  altcoin_max_position_value_ratio: number
  max_margin_usage: number
  min_position_size: number
  min_risk_reward_ratio: number
  min_confidence: number
}

export interface StrategyPromptSectionsConfigPayload {
  role_definition: string
  trading_frequency: string
  entry_standards: string
  decision_process: string
}

export interface StrategyGridConfigPayload {
  symbol: string
  grid_count: number
  total_investment: number
  leverage: number
  upper_price: number
  lower_price: number
  use_atr_bounds: boolean
  atr_multiplier: number
  distribution: string
  max_drawdown_pct: number
  stop_loss_pct: number
  daily_loss_limit_pct: number
  use_maker_only: boolean
  enable_direction_adjust: boolean
  direction_bias_ratio: number
}

export interface StrategySymbolConfig {
  symbol: string
  leverage: number
  min_cost?: number | null
  max_cost?: number | null
  fixed_cost?: number | null
}

export interface StrategyConfigPayload {
  symbols?: StrategySymbolConfig[]
  strategy_type?: string
  language?: string
  coin_source?: StrategyCoinSourceConfigPayload
  indicators?: StrategyIndicatorsConfigPayload
  custom_prompt?: string
  risk_control?: StrategyRiskControlConfigPayload
  prompt_sections?: StrategyPromptSectionsConfigPayload
  grid_config?: StrategyGridConfigPayload
  trading_symbols?: string
  max_positions?: number
  btc_eth_leverage?: number
  altcoin_leverage?: number
  prompt_variant?: string
}

export interface StrategyDecisionPayload {
  action: string
  symbol: string
  confidence: number
  reasoning: string
}

export interface CreateStrategyRequest {
  name: string
  description?: string
  config: StrategyConfigPayload
}

export interface UpdateStrategyRequest {
  name?: string
  description?: string
  config: StrategyConfigPayload
}

export interface DuplicateStrategyRequest {
  name?: string
}

export interface DefaultStrategyConfigQuery {
  lang?: string | null
}

export interface PreviewPromptRequest {
  config: StrategyConfigPayload
  account_equity?: number | null
  prompt_variant?: string | null
}

export interface StrategyTestRunRequest {
  config: StrategyConfigPayload
  prompt_variant?: string | null
  ai_model_id?: string | null
  run_real_ai?: boolean | null
}

export interface StrategyPayload {
  id: string
  name: string
  description: string
  author_email: string
  is_active: boolean
  is_default: boolean
  config: StrategyConfigPayload
  created_at: string
  updated_at: string
}

export interface StrategyListPayload {
  strategies: StrategyPayload[]
  count: number
}

export interface StrategyCreatedPayload {
  id: string
  message: string
}

export interface StrategyMessagePayload {
  message: string
}

export interface StrategyDefaultConfigPayload {
  language: string
  config: StrategyConfigPayload
}

export interface StrategyConfigSummaryPayload {
  coin_source: string
  primary_tf: string
  btc_eth_leverage: number
  altcoin_leverage: number
  max_positions: number
}

export interface PreviewPromptPayload {
  system_prompt: string
  prompt_variant: string
  config_summary: StrategyConfigSummaryPayload
}

export interface StrategyTestRunPayload {
  system_prompt: string
  user_prompt: string
  prompt_variant: string
  ai_model_id: string
  ai_response: string
  decisions: StrategyDecisionPayload[]
  reasoning: string
  duration_ms: number
  used_real_ai: boolean
}

File: core/src/contracts/strategies.rs
use serde::{Deserialize, Serialize};
use serde_json::Value;

#[derive(Debug, Deserialize)]
pub struct CreateStrategyRequest {
    pub name: String,
    #[serde(default)]
    pub description: String,
    pub config: Value,
}

#[derive(Debug, Deserialize)]
pub struct UpdateStrategyRequest {
    #[serde(default)]
    pub name: String,
    #[serde(default)]
    pub description: String,
    pub config: Value,
}

#[derive(Debug, Deserialize)]
pub struct DuplicateStrategyRequest {
    #[serde(default)]
    pub name: String,
}

#[derive(Debug, Deserialize)]
pub struct DefaultStrategyConfigQuery {
    pub lang: Option<String>,
}

#[derive(Debug, Deserialize)]
pub struct PreviewPromptRequest {
    pub config: Value,
    pub account_equity: Option<f64>,
    pub prompt_variant: Option<String>,
}

#[derive(Debug, Deserialize)]
pub struct StrategyTestRunRequest {
    pub config: Value,
    pub prompt_variant: Option<String>,
    pub ai_model_id: Option<String>,
    pub run_real_ai: Option<bool>,
}

#[derive(Debug, Clone, Serialize)]
pub struct StrategyPayload {
    pub id: String,
    pub name: String,
    pub description: String,
    pub author_email: String,
    pub is_active: bool,
    pub is_default: bool,
    pub config: Value,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize)]
pub struct StrategyListPayload {
    pub strategies: Vec<StrategyPayload>,
    pub count: usize,
}

#[derive(Debug, Clone, Serialize)]
pub struct StrategyCreatedPayload {
    pub id: String,
    pub message: &'static str,
}

#[derive(Debug, Clone, Serialize)]
pub struct StrategyMessagePayload {
    pub message: &'static str,
}

#[derive(Debug, Clone, Serialize)]
pub struct StrategyDefaultConfigPayload {
    pub language: String,
    pub config: Value,
}

#[derive(Debug, Clone, Serialize)]
pub struct StrategyConfigSummaryPayload {
    pub coin_source: String,
    pub primary_tf: String,
    pub btc_eth_leverage: i64,
    pub altcoin_leverage: i64,
    pub max_positions: i64,
}

#[derive(Debug, Clone, Serialize)]
pub struct PreviewPromptPayload {
    pub system_prompt: String,
    pub prompt_variant: String,
    pub config_summary: StrategyConfigSummaryPayload,
}

#[derive(Debug, Clone, Serialize)]
pub struct StrategyTestRunPayload {
    pub system_prompt: String,
    pub user_prompt: String,
    pub prompt_variant: String,
    pub ai_model_id: String,
    pub ai_response: String,
    pub decisions: Value,
    pub reasoning: String,
    pub duration_ms: u64,
    pub used_real_ai: bool,
}
